src/librustc_driver/lib.rs

```diff
@@ -1163,7 +1167,9 @@ pub fn version(binary: &str, matches: &getopts::Matches) {
         println!("commit-date: {}", unw(commit_date_str()));
         println!("host: {}", config::host_triple());
         println!("release: {}", unw(release_str()));
-        get_codegen_sysroot("llvm")().print_version();
+        let backend = option_env!("CFG_CODEGEN_BACKEND")
```

Comment:
So you turn on your backend with an environment variable? That's fine, but should be document it, at least in the README.md, or your own README.md?

Reply:
In a way, yes. However, you don't have to set `CFG_CODEGEN_BACKEND` manually. All you need to do is edit the `codegen-backends` line in `config.toml`: `codegen-backends = ["ironox"]`.

The reason this is necessary is because `./x.py build` run a piece of code that parses `config.toml`, and then runs `cargo build` on the rustc source. So, to communicate the backend specified in `config.toml` to rustc_driver, I've had to set the `CFG_CODEGEN_BACKEND` variable [here](https://github.com/gabi-250/rust/blob/27a8cc05b6e2b24d78e45d0c31c029ab09c6eb18/src/bootstrap/compile.rs#L545). I think this is the way you are 'supposed' to pass values from `config.toml` to the compiler, but I might be wrong.

Reply:
OK